src/components/element-editor/basic-details/basic-details.tsx

```diff
@@ -38,52 +36,57 @@ export const BasicDetails = () => {
     // =========================================================================
 
     return (
-        <div>
-            <Form.Input
-                {...register("id")}
-                required
-                label="ID"
-                placeholder="Enter ID"
-                errorMessage={errors.id?.message}
-            />
+        <FieldEditorAccordionItem type="default" expanded title="Basic">
+            <IconDropdown type={element?.type} id={element?.id} />
             {hasProperty("label") && (
-                <Form.Input
+                <Form.Textarea
                     required
                     label="Label"
-                    placeholder="Enter label"
+                    rows={1}
+                    placeholder="Element Name"
                     errorMessage={errors.label?.message}
+                    maxLength={40}
+                    value={element?.label}
                     {...register("label")}
                 />
             )}
-            {hasProperty("placeholder") && (
-                <Form.Input
-                    label="Placeholder"
-                    placeholder="Enter placeholder"
-                    errorMessage={errors.placeholder?.message}
-                    {...register("placeholder")}
-                />
-            )}
             <MandatoryFieldBox>
                 <Controller
                     name="required"
                     control={control}
-                    render={() => (
+                    defaultValue={true}
+                    render={({ field }) => (
                         <TogglePair
                             label="Mandatory field"
-                            onChange={(value) => setValue("required", value)}
+                            defaultValue={true}
+                            onChange={(value) => field.onChange(value)}
                         />
                     )}
                 />
-                {watch("required") && (
+                {watch("required", true) && (
                     <Form.Input
                         label="Error message"
+                        errorMessage={errors.requiredErrorMsg?.message}
                         {...register("requiredErrorMsg")}
                     />
                 )}
             </MandatoryFieldBox>
-            <Button.Small onClick={handleSubmit(handleClick)}>
-                Save
-            </Button.Small>
-        </div>
+            <Form.Input
+                {...register("id")}
+                required
+                label="ID"
+                placeholder="Enter ID"
+                errorMessage={errors.id?.message}
+            />
+
+            {hasProperty("placeholder") && (
+                <Form.Input
+                    label="Placeholder"
+                    placeholder="Enter placeholder text"
+                    errorMessage={errors.placeholder?.message}
+                    {...register("placeholder")}
+                />
+            )}
```

Comment:
there is a need to wrap each component in the `Controller` component in order to be integrated with react-hook-form, for more info, can refer to: https://react-hook-form.com/docs/usecontroller/controller